web/controllers/courseController/index.ts

```diff
@@ -1310,144 +995,75 @@ async function postLecture(student: Student, courseId: number, subcourseId: numb
  * @apiUse StatusInternalServerError
  */
 export async function putCourseHandler(req: Request, res: Response) {
-    let status = 204;
-    try {
-        if (res.locals.user instanceof Student) {
-            if (req.params.id != undefined &&
-                req.body.instructors instanceof Array &&
-                (req.body.name == undefined || typeof req.body.name == 'string') &&
-                (req.body.outline == undefined || typeof req.body.outline == 'string') &&
-                typeof req.body.description == 'string' &&
-                typeof req.body.allowContact === "boolean" &&
-                (req.body.correspondentID == undefined || typeof req.body.correspondentID === "string") &&
-                (req.body.outline == undefined || typeof req.body.category == 'string') &&
-                req.body.tags instanceof Array &&
-                (req.body.outline == undefined || typeof req.body.submit == 'boolean')) {
-
-                // Check if string arrays
-                for (let i = 0; i < req.body.instructors.length; i++) {
-                    if (typeof req.body.instructors[i] != 'string') {
-                        status = 400;
-                        logger.warn(`Instructor ID ${req.body.instructors[i]} is no string`);
-                    }
-                }
-                for (let i = 0; i < req.body.tags.length; i++) {
-                    if (typeof req.body.tags[i] != 'string') {
-                        status = 400;
-                        logger.warn(`Tag ID ${req.body.tags[i]} is no string`);
-                    }
-                }
+    handleError(res, async () => {
+        Validation.isStudent(res);
 
-                if (status < 300) {
-                    status = await putCourse(res.locals.user, Number.parseInt(req.params.id, 10), req.body);
-                }
-            } else {
-                status = 400;
-                logger.warn("Invalid request for PUT /course");
-                logger.debug(req.body);
-            }
-        } else {
-            status = 403;
-            logger.warn("A non-student wanted to edit a course");
-            logger.debug(res.locals.user);
+        Validation.hasParams(req, "id");
+        Validation.hasBody(req, { instructors: "string[]", description: "string", allowedContact: "boolean", name: "string?", outline: "string?", correspondentID: "string?", tags: "string[]" });
+        if (
+            (req.body.outline !== undefined && typeof req.body.category !== 'string') ||
+            (req.body.outline !== undefined && typeof req.body.submit !== 'boolean')) {
```

Comment:
@lsc0498 Do you know why this logic exists? Why does the outline have to not set the category? (This logic is equivalent in the pre-edit state)

Reply:
This doesn't make much sense. To me it looks much like a copy-paste error 😅

Reply:
Alright, removed that logic